odoo/addons/base/models/ir_actions_report_wkhtmltopdf.py

```diff
@@ -0,0 +1,391 @@
+import logging
+import lxml.html
+import os
+import re
+import subprocess
+import tempfile
+import threading
+import unittest
+
+from contextlib import closing, ExitStack
+from itertools import islice
+from lxml import etree
+from urllib.parse import urlparse
+
+from odoo import api, models, fields, tools, _
+from odoo.exceptions import UserError
+from odoo.http import request, root
+from odoo.service import security
+from odoo.tools.misc import find_in_path
+from odoo.tools import config, parse_version, split_every
+
+_logger = logging.getLogger(__name__)
+
+
+def _get_wkhtmltopdf_bin():
+    return find_in_path('wkhtmltopdf')
+
+
+def _get_wkhtmltoimage_bin():
+    return find_in_path('wkhtmltoimage')
+
+
+def _split_table(tree, max_rows):
```

Comment:
Since we are moving code, this could be in `odoo.tools.pdf`. This will simplify imports in test files as well.